pkg/common/common.go

```diff
@@ -80,8 +78,7 @@ func ReplaceManifestUnSupportedAPIs(origManifest, mapFile string, kubeConfig Kub
 		} else {
 			apiVersionStr = mapping.RemovedInVersion
 		}
-		apiVersion, err := strconv.ParseFloat(apiVersionStr, 32)
-		if err != nil {
+		if !semver.IsValid(apiVersionStr) {
 			errors.Wrapf(err, "Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI)
```

Comment:
errors.Wrapf(err, "Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI) --> errors.Errorf("Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI)

Reply:
also the output of errors.Errorf or Wrap should be handled.. 
right now it will be ignored AFAIK

Reply:
What behavior do you think would be best for the error in the for loop? Return immediately if an error is found from the map file, or wrap all and return the error following the loop?

Reply:
It looks like currently it wraps all errors but then the error isn't returned

Reply:
IMHO i would recomment to return it as this behaviour is also used a few lines above too.
so 
```
return "", errors.Errorf(...)
```

Reply:
Okay, I've changed error behavior to return immediately and improved the output formatting of the API details a bit